pkg/dhclient/dhclient.go

```diff
@@ -59,7 +59,10 @@ func IfUp(ifname string) (netlink.Link, error) {
 			return nil, fmt.Errorf("cannot get interface %q by name: %v", ifname, err)
 		}
 
-		if iface.Attrs().Flags&net.FlagUp == net.FlagUp {
+		// Check if link is actually operational.
```

Comment:
given that link up internally takes more than 30 seconds, do you want to pass the linkUpAttempt time in as a variable?